mountpoint-s3/src/main.rs

```diff
@@ -485,13 +490,94 @@ fn parse_perm_bits(perm_bit_str: &str) -> Result<u16, anyhow::Error> {
     }
 }
 
-pub fn retrieve_instance_type() -> anyhow::Result<String> {
+fn calculate_network_throughput() -> anyhow::Result<f64> {
+    let instance_type = retrieve_instance_type().context("failed to retrieve instance type")?;
+    let throughput = get_maximum_network_throughput(&instance_type).context("failed to get network throughput")?;
+    Ok(throughput)
+}
+
+fn retrieve_instance_type() -> anyhow::Result<String> {
     let imds_crt_client = ImdsCrtClient::new().context("failed to create IMDS client")?;
 
     let query = imds_crt_client
         .make_instance_type_query()
         .context("failed to send IMDS query")?;
 
     let result = futures::executor::block_on(query).context("IMDS query failed")?;
+    tracing::debug!("detected EC2 instance type {result}");
     Ok(result)
 }
+
+fn get_maximum_network_throughput(ec2_instance_type: &str) -> anyhow::Result<f64> {
+    const INSTANCE_THROUGHPUT: &str = "instance_throughput";
+    let file = include_str!("../scripts/network_performance.json");
+    let throughput: serde_json::Value =
+        serde_json::from_str(file).context("failed to parse network_performance.json")?;
+
+    throughput
+        .get(INSTANCE_THROUGHPUT)
+        .context("failed to parse instance throughput in network_performance.json")?
+        .get(ec2_instance_type)
+        .and_then(|t| t.as_f64())
+        .ok_or_else(|| anyhow!("no throughput configuration for EC2 instance type {ec2_instance_type}"))
```

Comment:
nit: `context(...)?` in the middle of a chain confuses me, so maybe it's clearer to lift the temporary out:

```suggestion
    let data: serde_json::Value = serde_json::from_str(file).context("failed to parse network_performance.json")?;

    let instance_throughput = data
        .get(INSTANCE_THROUGHPUT)
        .context("instance throughput missing from json")?;

    instance_throughput
        .get(ec2_instance_type)
        .and_then(|t| t.as_f64())
        .ok_or_else(|| anyhow!("no throughput configuration for EC2 instance type {ec2_instance_type}"))
```